File: main_project/AdHot/edm/AdNotifierI.cpp
#include "AdNotifierI.h"
#include "AdCache.h"
#include "ServiceI.h"

using namespace xce::ad;

void MyUtil::initialize() {
  ServiceI& service = ServiceI::instance();
  service.getAdapter()->add(&AdNotifierI::instance(), service.createIdentity("M", ""));
  CacheManager::instance().Init();
  TaskManager::instance().config(ThreadPoolConfig(1, 10));
  TaskManager::instance().scheduleRepeated(new ScanTask());
  int delay = ClearAdDailyPostedNumTask::DelayToXClock(1);
  TaskManager::instance().schedule(new ClearAdDailyPostedNumTask(delay));
  TaskManager::instance().schedule(new ClearUserWeeklyReceivedTask(5 * 60 * 1000)); // 五分钟后
}

void AdNotifierI::Login(int uid, const Ice::Current&) {
  TaskManager::instance().execute(new LoginTask(uid));
}

void AdNotifierI::Block(int uid, int promotion, const Ice::Current&) {
  MCE_INFO("AdNotifierI::Block uid=" << uid);
  TaskManager::instance().execute(new BlockTask(uid, promotion));
}

void AdNotifierI::Reload(const Ice::Current&) {
  CacheManager::instance().Init();
}

void AdNotifierI::SetInterval(int uid, IntervalType interval, const Ice::Current&) {
  return;
}

void AdNotifierI::MarkReadByUid(int uid, const Ice::Current&) {
  MCE_INFO("AdNotifierI::MarkReadByUid uid=" << uid);
  TaskManager::instance().execute(new MarkReadByUidTask(uid));
}

void AdNotifierI::LoadOnePromotion(int promotion, const Ice::Current&) {
  MCE_INFO("function LoadOnePromotion() outdated");
  return;
}

void AdNotifierI::DelPromotion(int promotion, const Ice::Current&) {
  TaskManager::instance().execute(new DelPromotionTask(promotion));
}

HotAdInfoSeq AdNotifierI::GetAdInfoSeq(const Ice::Current&) {
  return CacheManager::instance().GetAdInfoSeq();
}

bool AdNotifierI::ResetUserAd(Ice::Int uid, Ice::Int ad_id, const Ice::Current&) {
  return CacheManager::instance().ResetUserAd(uid, ad_id);
}

UserHotAdState AdNotifierI::GetUserAdState(Ice::Int uid, Ice::Int ad_id, const Ice::Current&) {
  return CacheManager::instance().GetUserAdState(uid, ad_id);
}

void LoginTask::handle() {
  CacheManager::instance().Login(uid_);
}

void BlockTask::handle() {
  CacheManager::instance().Block(uid_, promotion_);
}

void MarkReadByUidTask::handle() {
  CacheManager::instance().MarkReadByUid(uid_);
}

void DelPromotionTask::handle() {
  CacheManager::instance().DelAd(promotion_);
}

File: main_project/AdHot/notifyJ/src/main/java/com/renren/xce/impl/ad/AdNotifierI.java
package com.renren.xce.impl.ad;

import Ice.*;
import java.util.*;


import org.apache.log4j.Logger;

// import com.renren.xce.service.ad.AdHotPushService;

// import xce.ad.HotAdState;
// import xce.ad._AdNotifierOperations;
import xce.ad.AdNotifier;
import xce.ad.HotAdInfo;
import xce.ad.IntervalType;
import xce.ad.UserHotAdState;

// 应该是继承 AdNotifier
public final class AdNotifierI extends AdNotifier {

    private final Logger log = Logger.getLogger(AdNotifierI.class.getName());

    private final AdPushManager push_mgr_ = new AdPushManager();
    
    public AdNotifierI() {
		System.out.println("AdNotifierI");
    	push_mgr_.init();
    }
    
    @Override
    public Map<Integer, int[]> GetAdPartners(int uid, int[] ads, Current current) {
    	Map<Integer, int[]> ret = new HashMap<Integer, int[]>();
    	return ret;
    }

    @Override
    public void Login(int uid, Ice.Current __current) {
        log.info("Login uid=" + uid);
    	 push_mgr_.Login(uid);
    }

    @Override
    public void Block(int uid, int promotion, Ice.Current __current) {
    	
    }

    public void SetInterval(int uid, IntervalType time, Ice.Current __current) {
    	
    }

    @Override
    public void MarkReadByUid(int uid, Ice.Current __current) {
    	
    }

    @Override
    public void LoadOnePromotion(int promotion, Ice.Current __current) {
    	
    }

    @Override
    public void AddPromotion(int promotion, int[] partners, long starttime, long endtime, 
    		int level, boolean blockalways, Ice.Current __current) {
    	
    }

    @Override
    public void DelPromotion(int promotion, Ice.Current __current) {
    	
    }

    @Override
    public HotAdInfo[] GetAdInfoSeq(Ice.Current __current) {
    	return new HotAdInfo[0];
    }

    @Override
    public void Reload(Ice.Current current) {
    	
    }

    @Override
    public boolean ResetUserAd(int uid, int aid, Ice.Current __current) {
    	return false;
    }

    @Override
    public UserHotAdState GetUserAdState(int uid, int aid, Ice.Current __current) {
    	return new UserHotAdState();
    }
}
